core/crypto/abac/attributes.go

```diff
@@ -0,0 +1,267 @@
+/*
+Copyright IBM Corp. 2016 All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+		 http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package abac
+
+import (
+	"bytes"
+	"crypto/x509"
+	"encoding/asn1"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	pb "github.com/hyperledger/fabric/core/crypto/abac/proto"
+	"github.com/hyperledger/fabric/core/crypto/primitives"
+	"github.com/hyperledger/fabric/core/crypto/utils"
+
+	"github.com/golang/protobuf/proto"
+)
+
+var (
+	// TCertEncAttributesBase is the base ASN1 object identifier for attributes.
+	// When generating an extension to include the attribute an index will be
+	// appended to this Object Identifier.
+	TCertEncAttributesBase = asn1.ObjectIdentifier{1, 2, 3, 4, 5, 6}
+
+	// TCertAttributesHeaders is the ASN1 object identifier of attributes header.
+	TCertAttributesHeaders = asn1.ObjectIdentifier{1, 2, 3, 4, 5, 6, 9}
+
+	padding = []byte{255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255}
```

Comment:
When a value is decrypted the decryption does not failed if someone uses the wrong key. It just produce garbage information. The padding was introduced to allow someone to verify that the decryption was successful. It needs to be a constant known value.